src/Commands/Add.hs

```diff
@@ -0,0 +1,31 @@
+module Commands.Add
+    ( add
+    ) where
+
+
+import           Data.List        (lines, unlines)
+import           Entry
+import           Helpers          (getFilePath, indentedOutput, loadEntries,
+                                   nextRowNum, parseEntries, showOutput)
+import           System.Directory (removeFile, renameFile)
+import           System.IO        (readFile)
+
+
+add :: [String] -> IO ()
+add []              = putStrLn "No arguments provided.\nFor help: timetrack add --help"
+add ["-h"]          = putStrLn "usage: timetrack add YYYY-MM-DD \"message\""
+add ["--help"]      = putStrLn "usage: timetrack add YYYY-MM-DD \"message\""
+add [_]             = putStrLn "Not enough arguments provided.\nFor help: timetrack add --help"
+add [date, message] = do
+    path <- getFilePath
+    loadedLines <- loadEntries
+    let entries = parseEntries loadedLines
```

Comment:
Consider `entries <- fmap parseEntries loadEntries` and why it's equivalent.

Reply:
Because `IO` satisfies the `Functor` laws (identity and composition) and can be `map`ped over? So
```haskell
fmap :: (a -> b) -> f a -> f b
```
and in this case, `parseEntries` is taking some value of type `a` (a raw value here) and is returning some value of type `b` (an `Entry` here); the second parameter, `f a` is the `loadEntries` function that does its IO and returns us the `a` (raw string value here); we return `f b`, and _this seems to be where some magic is_... I'm assuming the bind here then extracts the value from the Functor and stores it in the variable `entries`?

Reply:
Yeah - it's often a convenient lie to think of a monadic value `m a` as a wrapper of type `m` containing a value of type `a`. If `action :: m a` then `x <- action` introduces a variable `x :: a`, which is like "unwrapping". 

(Thought exercise. Consider

```
putStrLn . show  $ do
  x <- a
  y <- b
  return (x, y)
```
for each of
```
a = Just 2, b = Just 3
a = Just 5, b = Nothing
a = [2,4,6], b = [1,3]
```
)

In this case specifically `m = IO` and as you noted, `fmap parseEntries :: IO [String] -> IO [Entry]` so `fmap parseEntries loadEntries :: IO [Entry]`.

Reply:
Thanks for taking the time to bang this out